server/src/internalClusterTest/java/org/opensearch/indices/CacheStatsAPIIndicesRequestCacheIT.java

```diff
@@ -183,20 +156,87 @@ public void testStatsMatchOldApi() throws Exception {
             0
         );
         // Don't check entries, as the old API doesn't track this
-        checkCacheStatsAPIResponse(xContentMap, xContentMapKeys, expected, true, false);
+        checkCacheStatsAPIResponse(statsHolder, List.of(), expected, true, false);
     }
 
     public void testNullLevels() throws Exception {
+        // Test the XContent in the response behaves correctly when we pass null levels.
         String index = "index";
         Client client = client();
         startIndex(client, index);
         int numKeys = Randomness.get().nextInt(100) + 1;
         for (int i = 0; i < numKeys; i++) {
             searchIndex(client, index, String.valueOf(i));
         }
-        Map<String, Object> xContentMap = getNodeCacheStatsXContentMap(client, null);
+        Map<String, Object> xContentMap = getStatsXContent(getNodeCacheStatsResult(client, null));
         // Null levels should result in only the total cache stats being returned -> 6 fields inside the response.
-        assertEquals(6, ((Map<String, Object>) xContentMap.get("request_cache")).size());
+        assertEquals(6, xContentMap.size());
+    }
+
+    public void testCacheClear() throws Exception {
+        // Cache clear API should clear memory size and number of items, but not other stats
+        String index = "index";
+        Client client = client();
+
+        startIndex(client, index);
+
+        int expectedHits = 2;
+        int expectedMisses = 7;
+        // Search for the same doc to give hits
+        for (int i = 0; i < expectedHits + 1; i++) {
+            searchIndex(client, index, "");
+        }
+        // Search for new docs
+        for (int i = 0; i < expectedMisses - 1; i++) {
+            searchIndex(client, index, String.valueOf(i));
+        }
+
+        ImmutableCacheStats expectedTotal = new ImmutableCacheStats(expectedHits, expectedMisses, 0, 0, expectedMisses);
+        ImmutableCacheStatsHolder statsHolder = getNodeCacheStatsResult(client, List.of());
+        // Don't check the memory size, just assert it's nonzero
+        checkCacheStatsAPIResponse(statsHolder, List.of(), expectedTotal, false, true);
+        long originalMemorySize = statsHolder.getTotalSizeInBytes();
+        assertNotEquals(0, originalMemorySize);
+
+        // Clear cache
+        ClearIndicesCacheRequest clearIndicesCacheRequest = new ClearIndicesCacheRequest(index);
+        client.admin().indices().clearCache(clearIndicesCacheRequest).actionGet();
+
+        // Now size and items should be 0
+        expectedTotal = new ImmutableCacheStats(expectedHits, expectedMisses, 0, 0, 0);
+        statsHolder = getNodeCacheStatsResult(client, List.of());
+        checkCacheStatsAPIResponse(statsHolder, List.of(), expectedTotal, true, true);
+    }
+
+    public void testNodesStatsResponse() throws Exception {
```

Comment:
Can you add detailed description regarding the specific test case covered above each test method?